File: backend/app/schemas/translate.py
from pydantic import BaseModel, Field
from datetime import datetime
from decimal import Decimal
from typing import Optional


class TranslateRequest(BaseModel):
    source_lang: str = Field(..., min_length=2, max_length=10, examples=["ko"])
    target_lang: str = Field(..., min_length=2, max_length=10, examples=["en"])
    text: str = Field(..., min_length=1, max_length=5000)
    input_type: str = Field(default="text", max_length=10)
    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None
    country_code: Optional[str] = Field(default=None, max_length=5)


class TranslateResponse(BaseModel):
    source_lang: str
    target_lang: str
    source_text: str
    translated_text: str


class TranslationRecord(BaseModel):
    id: int
    source_lang: str
    target_lang: str
    source_text: str
    translated_text: str
    input_type: str
    latitude: Optional[Decimal] = None
    longitude: Optional[Decimal] = None
    country_code: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/translate.ts
import axios from "axios";

const api = axios.create({
  baseURL: "/api/v1",
  headers: { "Content-Type": "application/json" },
});

export interface TranslateRequest {
  source_lang: string;
  target_lang: string;
  text: string;
}

export interface TranslateResponse {
  source_lang: string;
  target_lang: string;
  source_text: string;
  translated_text: string;
}

export interface TranslationRecord {
  id: number;
  source_lang: string;
  target_lang: string;
  source_text: string;
  translated_text: string;
  input_type: "text" | "voice" | "document" | "website";
  latitude: number | null;
  longitude: number | null;
  country_code: string | null;
  created_at: string;
}

export async function postTranslate(
  req: TranslateRequest
): Promise<TranslateResponse> {
  const { data } = await api.post<TranslateResponse>("/translate/", req);
  return data;
}

export async function getTranslateHistory(skip: number = 0, limit: number = 10): Promise<TranslationRecord[]> {
  const { data } = await api.get<TranslationRecord[]>("/translate/history", {
    params: { skip, limit },
  });
  return data;
}